primitives/transaction-pool/src/pool.rs

```diff
@@ -225,6 +226,13 @@ pub trait TransactionPool: Send + Sync {
 	fn hash_of(&self, xt: &TransactionFor<Self>) -> TxHash<Self>;
 }
 
+/// Trait for transaction pool maintaince.
+pub trait MaintainedTransactionPool : TransactionPool {
```

Comment:
Do we still need this separate trait? Perhaps better to just merge with `TransactionPool`?

Reply:
@svyatonik thinks we need it below

Reply:
I don't have any specific opinion on whether we need this or not :) I just want my feature back! :)

Reply:
Let's just keep `TransactionPoolMaintainer` trait then and just implement it for `BasePool` by default.
If other maintainance is needed, one can:
1. Create a wrapper struct.
2. Implement `TransactionPool` and delegate to the original one
3. Implement different `TransactionPoolMaintainer`.

I like the fact that implementing the maintainer directly on the pool significantly reduces the amount of code and allows the maintainer to use internal methods. Maybe I'm still to be convinced on possible use cases of customizing only the maintainer, but from my POV it's not going to be common and I'm willing to trade-off easy of replacing with less code to maintain on our side.

Reply:
@tomusdrw yeah my thoughts exactly

I don't think we need to keep bunch of confusing code just in case somebody may need it

Reply:
> I don't have any specific opinion on whether we need this or not :) I just want my feature back! :)

Feature is there I think.
Start using it is not much more difficult that it would have been before :)

Reply:
@svyatonik and btw, I will be doing `FullPull` that will contain custom maintaince in next PR and address your concerns there, ok?

Reply:
I don't think I'll have any concerns other than supporting the custom maintenance algorithm right now, neither I have idea of what do you mean by `FullPull` :) So as I understand, we're doing something like that:
(1) `struct MyPool { inner: txpool::Pool<PoolApi> }`
(2) `impl TransactionPool for MyPool { delegate-everything-to-inner }`
(3) `impl MaintainedTransactionPool for MyPool { custom-logic }`

Looks good to me. I'll take a look at light-related code now

Reply:
cool,
(1) can be just `sc_transaction_graph:Pool`
BasicPool is already just `sc_transaction_graph:Pool` + maintaince